File: sdk/python/examples/list_tickets.py
"""List one bounded page without printing credentials or ticket content."""

from __future__ import annotations

import os

from chronodesk import Audience, ChronoDeskClient, ClientCredentials


def main() -> None:
    project_key = required_environment("CHRONODESK_PROJECT_KEY")
    anonymous = ChronoDeskClient(
        required_environment("CHRONODESK_URL"),
        project_key,
    )
    token = anonymous.exchange_client_credentials(
        ClientCredentials(
            client_id=required_environment("CHRONODESK_CLIENT_ID"),
            client_secret=required_environment("CHRONODESK_CLIENT_SECRET"),
            audience=Audience.API,
            scopes=("tickets:read",),
        )
    )
    result = anonymous.with_access_token(token.access_token).list_tickets(limit=20)
    print(
        f"project={project_key} tickets={len(result['data'])} "
        f"request_id={result['meta']['request_id']}"
    )


def required_environment(name: str) -> str:
    value = os.environ.get(name, "")
    if not value:
        raise RuntimeError(f"required environment variable {name} is not set")
    return value


if __name__ == "__main__":
    main()

File: sdk/typescript/examples/list-tickets.ts
import { ChronoDeskClient } from "../src/index.js";

const environment = globalThis as typeof globalThis & {
  process?: { env?: Record<string, string | undefined> };
};

const projectKey = requiredEnvironment("CHRONODESK_PROJECT_KEY");
const anonymous = new ChronoDeskClient(
  requiredEnvironment("CHRONODESK_URL"),
  projectKey,
);
const token = await anonymous.exchangeClientCredentials({
  clientId: requiredEnvironment("CHRONODESK_CLIENT_ID"),
  clientSecret: requiredEnvironment("CHRONODESK_CLIENT_SECRET"),
  audience: "api",
  scopes: ["tickets:read"],
});
const result = await anonymous
  .withAccessToken(token.access_token)
  .listTickets({ limit: 20 });
console.log(
  `project=${projectKey} tickets=${result.data.length} ` +
    `request_id=${result.meta.request_id}`,
);

function requiredEnvironment(name: string): string {
  const value = environment.process?.env?.[name];
  if (!value) {
    throw new Error(`required environment variable ${name} is not set`);
  }
  return value;
}
